EIPS/eip-5334.md

```diff
@@ -0,0 +1,270 @@
+---
+eip: 5334
+title: 721 User And Expires And Level Extension
+description: Add a time-limited role with restricted permissions to ERC-721 tokens.
+author: Yan (@yan253319066)
+discussions-to: https://ethereum-magicians.org/t/erc-721-user-and-expires-and-level-extension/10097
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-07-25
+requires: 165, 721
+---
+
+## Abstract
+
+This standard is an extension of [EIP-721](./eip-721.md). It proposes an additional role (`user`) which can be granted to addresses, and a time where the role is automatically revoked (`expires`) and (`level`) . The `user` role represents permission to "use" the NFT, but not the ability to transfer it or set users.
+
+## Motivation
+
+Some NFTs have certain utilities. For example, virtual land can be "used" to build scenes, and NFTs representing game assets can be "used" in-game. In some cases, the owner and user may not always be the same. There may be an owner of the NFT that rents it out to a “user”. The actions that a “user” should be able to take with an NFT would be different from the “owner” (for instance, “users” usually shouldn’t be able to sell ownership of the NFT).  In these situations, it makes sense to have separate roles that identify whether an address represents an “owner” or a “user” and manage permissions to perform actions accordingly.
+
+Some projects already use this design scheme under different names such as “operator” or “controller” but as it becomes more and more prevalent, we need a unified standard to facilitate collaboration amongst all applications.
+
+Furthermore, applications of this model (such as renting) often demand that user addresses have only temporary access to using the NFT. Normally, this means the owner needs to submit two on-chain transactions, one to list a new address as the new user role at the start of the duration and one to reclaim the user role at the end. This is inefficient in both labor and gas and so an “expires” and “level” function is introduced that would facilitate the automatic end of a usage term without the need of a second transaction.
+
+## Specification
+
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY" and "OPTIONAL" in this document are to be interpreted as described in RFC 2119.
+
+### Contract Interface
+Solidity Interface with NatSpec & OpenZeppelin v4 Interfaces (also available at [IERC5334.sol](../assets/eip-5334/contracts/IERC5334.sol)):
```

Comment:
@SamWilsn It would be nice if the bot didn't complain about this here:
> proposals must be referenced with the form `EIP-N` (not `ERC-N`)

Reply:
Wrap it in code blocks
```suggestion
Solidity Interface with NatSpec & OpenZeppelin v4 Interfaces (also available at [`IERC5334.sol`](../assets/eip-5334/contracts/IERC5334.sol)):
```